stdlib/Distributed/src/remotecall.jl

```diff
@@ -247,22 +247,32 @@ function del_clients(pairs::Vector)
     end
 end
 
-const any_gc_flag = Condition()
+const any_gc_flag = Threads.Condition()
 function start_gc_msgs_task()
-    @async while true
-        wait(any_gc_flag)
-        flush_gc_msgs()
+    @async begin
+        lock(any_gc_flag) do
+            wait(any_gc_flag)
+            flush_gc_msgs()
+        end
     end
 end
 
+# Function can be called within a finalizer
 function send_del_client(rr)
     if rr.where == myid()
         del_client(rr)
     elseif id_in_procs(rr.where) # process only if a valid worker
         w = worker_from_id(rr.where)
-        push!(w.del_msgs, (remoteref_id(rr), myid()))
-        w.gcflag = true
-        notify(any_gc_flag)
+        msg = (remoteref_id(rr), myid())
+        Threads.@spawn begin # Need to spawn task since we can be in finalizer
```

Comment:
I've worked with Go some time ago and I've learned to appreciate that all errors/failures from goroutines surface eventually. Is there some structure to catch errors in these tasks whose references have been lost (gone out of scope)?